src/biokbase/narrative/tests/test_job.py

```diff
@@ -780,8 +811,70 @@ def test_refresh_attrs__batch(self):
         self.check_job_attrs(job, job_id)
 
         def mock_check_job(self_, params):
-            assert params["job_id"] == job_id
+            self.assertEqual(params["job_id"], job_id)
             return {"child_jobs": self.NEW_CHILD_JOBS}
 
         with mock.patch.object(MockClients, "check_job", mock_check_job):
             self.check_job_attrs(job, job_id, {"child_jobs": self.NEW_CHILD_JOBS})
+
+    def test_was_terminal(self):
+        all_jobs = get_all_jobs()
+
+        for job_id, job in all_jobs.items():
+            self.assertEqual(
+                JOBS_TERMINALITY[job_id],
+                job.was_terminal
+            )
+
+    @mock.patch("biokbase.narrative.jobs.job.clients.get", get_mock_client)
+    def test_was_terminal__batch(self):
+        batch_fam = get_batch_family_jobs(return_list=True)
+        batch_job, child_jobs = batch_fam[0], batch_fam[1:]
+
+        self.assertFalse(batch_job.was_terminal)
+
+        def mock_check_job(self_, params):
+            assert params["job_id"] in BATCH_CHILDREN
+            return {"status": COMPLETED_STATUS}
+
+        with mock.patch.object(MockClients, "check_job", mock_check_job):
+            for job in child_jobs:
+                job.state(force_refresh=True)
+
+        self.assertTrue(batch_job.was_terminal)
+
+    def test_in_cells(self):
+        all_jobs = get_all_jobs()
+        cell_2_jobs = get_cell_2_jobs(instance=False)
+        cell_ids = list(cell_2_jobs.keys())
+        # Iterate through all combinations of cell IDs
+        for combo_len in range(len(cell_ids) + 1):
+            for combo in itertools.combinations(cell_ids, combo_len):
+                combo = list(combo)
+                # Get jobs expected to be associated with the cell IDs
+                exp_job_ids = [
+                    job_id
+                    for cell_id, job_ids in cell_2_jobs.items()
+                    for job_id in job_ids
+                    if cell_id in combo
+                ]
+                for job_id, job in all_jobs.items():
+                    self.assertEqual(
+                        job_id in exp_job_ids,
+                        job.in_cells(combo)
+                    )
+
+    def test_in_cells__batch(self):
+        batch_fam = get_batch_family_jobs(return_list=True)
+        batch_job, child_jobs = batch_fam[0], batch_fam[1:]
+
+        for job in child_jobs:
+            job.cell_id = "hello"
```

Comment:
Could you also add some tests for cases where the child jobs have different `cell_id`s?

Reply:
Good catch! There was a bug